tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1880,6 +1892,240 @@ tensorflow::Status ConvertReshape(OpConverterParams* params) {
+        (output_bound[i] > input_dims[i])) {
+      return tensorflow::errors::InvalidArgument(
+          bound_name,
+          " for StridedSlice is invalid, must be in the range "
+          "[-rank(input), rank(input)], at ",
+          node_name);
+    }
+    // Convert negative values to their positive equivalent.
+    if (output_bound[i] < 0) {
+      output_bound[i] += input_dims[i];
+    }
+  }
+  return tensorflow::Status::OK();
+}
+
+tensorflow::Status ConvertStridedSlice(OpConverterParams* params) {
+  const auto& inputs = params->inputs;
+  const auto& node_def = params->node_def;
+  if (inputs.size() != 4) {
+    return tensorflow::errors::InvalidArgument(
+        "StridedSlice expects 4 inputs, at ", node_def.name());
+  }
+  if (!inputs.at(1).is_weights() || !inputs.at(2).is_weights() ||
+      !inputs.at(3).is_weights()) {
+    return tensorflow::errors::InvalidArgument(
+        "StridedSlice expects weights for begin, end, and strides, at ",
+        node_def.name());
+  }
+  if (!inputs.at(0).is_tensor()) {
+    return tensorflow::errors::Unimplemented(
+        "StridedSlice is only implemented for tensors, at ", node_def.name());
+  }
+  // Get input dims.
+  nvinfer1::Dims dims = inputs.at(0).GetTrtDims();
+  std::vector<int> input_dims(dims.d, dims.d + dims.nbDims);
+  if (inputs.at(0).is_tensor()) {
+    // Temporarily add batch dimension so that indexes line up properly.
+    input_dims.insert(input_dims.begin(), inputs.at(0).batch_size());
+  }
+  if (input_dims.size() > 4) {
+    return tensorflow::errors::Unimplemented(
+        "StridedSlice is not implemented for tensors with rank > 4, at ", 
+        node_def.name());
+  }
+  TFAttrs attrs(node_def);
+  // Get begin and end bounds per axis.
+  std::vector<int> begin, end;
+  TF_RETURN_IF_ERROR(GetStridedSliceBound(input_dims, inputs.at(1).weights(),
+                                          "begin", node_def.name(), begin));
+  TF_RETURN_IF_ERROR(GetStridedSliceBound(input_dims, inputs.at(2).weights(),
+                                          "end", node_def.name(), end));
+  int begin_mask = attrs.get<int>("begin_mask");
+  for (int i = 0; i < begin.size(); i++) {
+    if ((1 << i) & begin_mask) {
+      begin[i] = 0;
+    }
+  }
+  int end_mask = attrs.get<int>("end_mask");
+  for (int i = 0; i < end.size(); i++) {
+    if ((1 << i) & end_mask) {
+      end[i] = input_dims[i];
+    }
+  }
+  // Get strides per axis (must all be 1).
+  TRT_ShapedWeights stride_weights = inputs.at(3).weights();
+  const int* stride_weights_ptr =
+      static_cast<int*>(const_cast<void*>(stride_weights.GetValues()));
+  std::vector<int> strides(stride_weights_ptr,
+                           stride_weights_ptr + stride_weights.count());
+  for (int x : strides) {
+    if (x != 1) {
+      return tensorflow::errors::Unimplemented(
+          "StridedSlice is only implemented for stride of 1, at ", 
+          node_def.name());
+    }
+  }
+  // Unsupported options.
```

Comment:
Isn't this comment a bit misleading? There are other unsupported options in the code above this.